Decisions/scs-XXXX-v1-requirements-for-sso-identity-federation.md

```diff
@@ -0,0 +1,187 @@
+3rd party python module as well as ansible support for it. Both of these are
+downstream of Keycloak itself and may thus not always be feature complete and
+suffer latency with respect to getting adjusted to upstream changes.
+
+It offers support for commonly used SSO protocols and is "reasonably" fast
+in adopting to protocol standard changes and extensions. This has been
+observed in the case of logout support (backend and frontend variants) in OIDC.
+
+It offers a concept of "Identity Brokering", where Keycloak is not just IdP
+but also "client" to other IdPs. This allows daisy chaining of identity
+federation. In this configuration it can work as a point of protocol
+transition between different supported SSO protocols (SAML, OAuth 2.0, etc.).
+
+Beyond this capability of using other IdPs as identity sources, it also supports
+using classic LDAP based IAM services as backend (OpenLDAP and Active Directory,
+e.g.).
+
+Keycloaks implementation makes some design decisions, that are specific
+to it and have consequences for clients of the service. E.g. Keycloak
+has a concept of management "Realms", which have their own specific
+set of HTTP API entrypoints, both for administration as well as for IdP
+requests.
+
+Commonly Keycloak realms can be used to map them 1:1 to user domains,
+but since Keycloak supports configuring multiple backend IdPs in a
+realm to be used for "Identity Brokering", there is always the
+possibility to create a kind of "proxy" realm to provide a single
+standard set of HTTP API endpoints for SSO clients (service providers)
+to avoid the need to frequently extend/reduce client service configuration
+whenever a new IdP federation needs to be added to Keycloak to onboard
+a new customer. This is relevant for services like OpenStack Keystone,
+which currently cannot be easily reconfigured for new SSO endpoints
+without restarting the service, making the service unavailable for
+a short span of time and increasing risk connected with service restarts.
+
+Since version 17, Keycloak claims that it's capability for
+"cloud neative" deployments on Kubernetes has improved.
+
+[Keycloak is offering a REST API](https://www.keycloak.org/docs-api/20.0.1/rest-api/index.html)
+for all aspects of its administration interface.
+
+For storage of Keycloak configuration and local user metadata
+(e.g. from which external IdP a user account originally came from)
+Keycloak supports several SQL backends through JDBC. Thus
+it can be hooked up to a Postgres Database or to a
+MariaDB/Galera cluster e.g..
+
+Keycloak is currently beeing deployed as part of the OSISM testbed.
+Technically this IdP component shall be shifted from the management
+plane to be run on the basis of a "minimal" Kubernetes (e.g. K3S),
+e.g. to make use of the "self healing" and scaling features achievable
+with that. As a central service for identity handling, the IdP
+service needs to be realy really robust.
```

Comment:
This is a repetition of the same sentence above.

*Maybe* Kubernetes's features could also be omitted, as they are not too relevant.

Reply:
I think it is a relevant aspect for the selection of an IdP solution: Does it work well on a K8S environment or is it maybe not designed for that at all? The instances will need to share configuration (probably via the shared backend database) as well as session state. Maybe some are better prepared for horizontal scaling than others.

Reply:
Agreed. Maybe just rephrase it exactly like that.
And also move it into the requirements/"design considerations" section or give Zitadel a matching paragraph as well?

Reply:
Good point, yeah, I was too much in "Brainstorm" mode when writing about the specific options. I'll adjust this.

Reply:
Adjusted in latest commit.